Adding `ard_strata()`

ddsjoberg opened the pull request:
**What changes are proposed in this pull request?**
* Style this entry in a way that can be copied directly into `NEWS.md`. (#<issue number>, @<username>)

Provide more detail here as needed.

**Reference GitHub issue associated with pull request.** _e.g., 'closes #<issue number>'_
closes #273

--------------------------------------------------------------------------------

Pre-review Checklist (if item does not apply, mark is as complete)
- [ ] **All** GitHub Action workflows pass with a :white_check_mark:
- [ ] PR branch has pulled the most recent updates from master branch: `usethis::pr_merge_main()`
- [ ] If a bug was fixed, a unit test was added.
- [ ] Code coverage is suitable for any new functions/features (generally, 100% coverage for new code): `devtools::test_coverage()`
- [ ] Request a reviewer

Reviewer Checklist (if item does not apply, mark is as complete)

- [ ] If a bug was fixed, a unit test was added.
- [ ] Run `pkgdown::build_site()`. Check the R console for errors, and review the rendered website.
- [ ] Code coverage is suitable for any new functions/features: `devtools::test_coverage()`

When the branch is ready to be merged:
- [ ] Update `NEWS.md` with the changes from this pull request under the heading "`# cards (development version)`". If there is an issue associated with the pull request, reference it in parentheses at the end update (see `NEWS.md` for examples).
- [ ] **All** GitHub Action workflows pass with a :white_check_mark:
- [ ] Approve Pull Request
- [ ] Merge the PR. Please use "Squash and merge" or "Rebase and merge".

bzkrouse reviewed (changes requested):
This looks really great!! I played around a bit and noticed a couple of very minor things unrelated to the core functionality:
1. if a group/strata appears in the `.f` function, then the order of the group columns (group1, group1_level, group2, group2_level) is different than the typical order of ard_* functions
2. if the `.f` function contains more than 1 group/strata then error is thrown (`ard_strata(ADSL, .by = ARM, .f = ~ ard_continuous(.x, by = c(SEX, AGEGR1), variables = AGE))`)
3. if ard_strata has no `.by` and the `.f` function contains a group/strata then an error is thrown (`ard_strata(ADSL, .f = ~ ard_continuous(.x, by = "ARM",variables = AGE))`)

ddsjoberg commented:
Oh goodness, thank you for the thorough review!!

> 1. if a group/strata appears in the `.f` function, then the order of the group columns (group1, group1_level, group2, group2_level) is different than the typical order of ard_* functions

I added a note in the help file for the `.by` and `.strata` args not to include columns that appear in the function.

> 2. if the `.f` function contains more than 1 group/strata then error is thrown (`ard_strata(ADSL, .by = ARM, .f = ~ ard_continuous(.x, by = c(SEX, AGEGR1), variables = AGE))`)

Ah, thank you! Updated!

> 3. if ard_strata has no `.by` and the `.f` function contains a group/strata then an error is thrown (`ard_strata(ADSL, .f = ~ ard_continuous(.x, by = "ARM",variables = AGE))`)

Ah, thank you x2! Updated!
